pkg/driver/agent.go

```diff
@@ -53,6 +54,14 @@ func NewNode(d *CSIDriver) csi.NodeServer {
 	// set up signals so we handle the first shutdown signal gracefully
 	stopCh := signals.SetupSignalHandler()
 
+	// start the lvm node resource watcher
```

Comment:
why do we need to have controller for lvm node, there is no need of having watcher for the object.

Reply:
Yes, it's possible to keep lvm node resource in sync without the controller. But there are few things I had in mind at that time before taking up the controller path. Informers are anyways required for upserting the lvmnode objects. In absence  of informers, we'll end up querying api server (thus increasing load on control plane) for every periodic update since we need to first fetch and then update the resource if required. It'll becomes more visible in case of aggressive sync period say 5s in large clusters. Only extension I can think of between controllers and informers is usage of worker queue to process each event on resource (including periodic resync). Now, currently we reconcile lvm node object almost immediately in case admin updates/deletes that resource by mistake rather than waiting for next sync period. (As I can now, the [CSIStorageCapacity](https://github.com/kubernetes-csi/external-provisioner/blob/master/pkg/capacity/capacity.go#L271) also uses similar controller based approach to reconcile immediately)

Since all the other CRDs (lvm volume & lvm snapshot) are building using the similar pattern, I thought of using the same   constructs for better understandability. But I agree, we could have used just informers cache and write a loop which updates these lvmnode resources periodically. I'm wondering do we see any specific issue with this controller approach?

Reply:
> we could have used just informers cache and write a loop which updates these lvmnode resources periodically

yes, it makes sense to use informer cache and a write loop. watcher does not make any sense here as node daemonset does not have to do anything in response to update/delete of the lvmnode resource. Can we take this as enhancement in the next release, I will go ahead and merge this PR?

Reply:
Sure. We can go ahead with current implementation & discuss the enhancement in next meeting. For curiosity purpose, I see the [CSIStorageCapacity](https://github.com/kubernetes-csi/external-provisioner/blob/master/pkg/capacity/capacity.go#L271) (a very similar use case as ours) resources managed by external provisioner also reconcile immediately in case of any rouge update or deletion of these objects by say cluster admin. I think, with reconciling immediately we are just reducing the blast radius of stale information (although staleness is possible with other scenarios). Anyways, let's discuss this in next meeting.